src/components/Graph/Graph.tsx

```diff
@@ -223,11 +241,12 @@ const GraphMemo: FC<GraphMemoProps> = memo(
                     nodeRefs.current[nodeName] = node;
                   }}
                   pathwayState={pathway.states[nodeName]}
-                  isCurrentNode={false}
+                  isCurrentNode={pathway.states[nodeName].label === currentNode?.label}
```

Comment:
As you mentioned in the comments the label is not unique but the key is. Since we are dealing with the entire nodes here and not just passing around the label I think the only change needed is:
```suggestion
                  isCurrentNode={pathway.states[nodeName].key === currentNode?.key}
```